HW04/Vadim_Volovenko/implementations_spec.rb

```diff
@@ -0,0 +1,51 @@
+# rubocop:disable Metrics/BlockLength
+require 'rspec'
+require_relative 'implementations.rb'
+
+RSpec.describe Implementations do
+  subject { described_class.new([1, 2, 3]) }
+
+  describe '#valid?' do
+    context 'when array is not empty' do
+      it 'array valid' do
+        expect(subject.valid?).to eq true
+      end
+    end
+
+    context 'when array empty' do
+      subject { described_class.new([]) }
+
+      it 'array nill' do
+        expect(subject.valid?).to eq false
+      end
+    end
+  end
+
+  describe '#my_map' do
+    context 'when array valid' do
+      it 'return right array' do
```

Comment:
Your spec description should look like documentation. Be explicit describing your spec, what is a valid array?  What is the right array? 

should be something like this 
```suggestion
    context 'when array is present' do
      it 'transforms an array based on block conditions' do
```